pygments/lexers/c_cpp.py

```diff
@@ -174,6 +178,13 @@ class CFamilyLexer(RegexLexer):
             # template specification
             (r'\s*(?=>)', Text, '#pop'),
             default('#pop')
+        ],
+        # Mark identifiers preceded by `case` keyword as constants.
+        'case-value': [
+            (r'(:)(?!:)', Punctuation, '#pop'),
```

Comment:
Sadly, it's a bit more complicated than that:

```c
  const int a = 1;
  const int b = 2;

  switch (4)
  case (a ? a : b):
  ...
```

Reply:
Thank you for the quick review!

While this approach isn't perfect, it will rarely fail, with convoluted code that is rarely written. I think that in languages that are hard to lex, like C++, a simple and decent solution should be considered, even if it's not perfect.

I searched for `case` keyword usage in [Juce's source code](https://github.com/juce-framework/JUCE/) (big open source C++ GUI framework, with it's own standard library). After looking at half of all the usages, I failed to see one which is not handled correctly by the proposed solution.

This is just my opinion though, and I am happy to make modifications.

Reply:
In this case, I think it should not require very complex logic to fix the issue, with something like (untested)

```python
'case-value': [
    (r':(?!:)', Punctuation, '#pop'),
    (r'\(', Punctuation, 'case-expr'),
],
'case-expr': [
    (r'\(', Punctuation, '#push'),
    (r'\)', Punctuation, '#pop'),
    (_namespaced_ident, Name.Constant),
    include('whitespace'),
    include('statements'),
]
```

I agree that lexers needn't be perfect, the fix for this issue just seemed acceptably simple (and the languages common enough) for it to be worth it (maybe I'm missing something though).

Reply:
That's also not perfect, as the ternary operator doesn't have to be enclosed with parentheses  -- see [this godbolt link](https://godbolt.org/z/YxzY4TxsM) for an example.

I appreciate the help and direction given towards a solution. Unfortunately, a stray colon can appear after a `case` in many ways. This is because since C++17, lambdas are constant expressions, so an immediately invoked lambda can follow a `case` keyword, and inside this lambda, pretty much every syntax can appear.

I created a [short example](https://godbolt.org/z/r4jEavz3j) showcasing this. Colons appear in:

- Line 6 (class definition).
- Line 7 (access modifier).
- Line 8 (bit field).
- Line 12 (ranged for loop).

I think this is a lost battle :(

Reply:
Ugh, OK. Somehow I was under the impression that the case statement required parentheses (not a C/C++ expert). LGTM then.